src/main/java/com/org/trustservice/controller/TrustGroupPermissionController.java

```diff
@@ -0,0 +1,63 @@
+package com.org.trustservice.controller;
+
+import com.org.trustservice.dto.TrustGroupUpdateDTO;
+import com.org.trustservice.dto.TrustGroupUpdateResponseDTO;
+import com.org.trustservice.excpetion.GenericAPIException;
+import com.org.trustservice.service.TrustGroupService;
+import com.org.trustservice.util.Constants;
+import com.org.trustservice.util.ControllerResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.context.request.WebRequest;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/api/v1")
+public class TrustGroupPermissionController {
+
+    @Autowired
+    TrustGroupService trustGroupService;
+
+
+    @GetMapping("/departments/{departmentId}/trustGroups")
+    public ResponseEntity<TrustGroupUpdateResponseDTO> getTrustGroupsByDepartment(
+            @PathVariable("departmentId") UUID departmentId,
+            WebRequest request) {
+        TrustGroupUpdateResponseDTO responseDTO = null;
+        try {
+            if (!isOrganizationDetailsExistsInHeader(request)) {
+                throw new GenericAPIException("Organization details missing in header");
+            }
+            responseDTO = trustGroupService.getAllTrustGroupsByDepartment(
+                    UUID.fromString(request.getHeader(Constants.ORG_COLLAB_HEADER)),
+                    departmentId);
+        } catch (Exception e) {
+            throw new GenericAPIException("Something went wrong ::" + e.getLocalizedMessage(), e);
+        }
+        return ControllerResponse.getOkResponseEntity(responseDTO);
```

Comment:
By default, returning an object means `OK` response. It doesn't need to be wrapped in ResponseEntity specially when the code is throwing and exception for failed case.
Example1:
```java
if(solution != null){
    return solution; //No wrapping in ResponseEntity
}else{
    throw Exception;
}
``` 

Example2
```java
if(solution.isValid()){
    return new ResponseEntity<>(solution, HttpStatus.OK);
}else{
    return new ResponseEntity<>(solution, HttpStatus.BAD_REQUEST);
}
```